docs/source/guides/error_handling.rst

```diff
@@ -0,0 +1,118 @@
+that extend from the WebApplicationException hierarchy. This includes exceptions such as
+BadRequestException, NotFoundException, custom exceptions extending from WebApplicationException,
+etc. WebApplicationExceptionMapper treats the exception as-is but sanitizes the exception's
+response, status, and message and formats it as JSON before sending back a response. The status
+will be propagated through to the response as long as it is a valid code; if the code is invalid,
+this will be treated as an Internal Server Error (500). The response message will be derived from
+the resulting status code.
+
+The FailureExceptionMapper is meant to be used with projects using Resteasy and handles all
+exceptions thrown internally by Resteasy that extend from the Failure hierarchy. Resteasy has
+`built-in internally-thrown exceptions <https://docs.jboss.org/resteasy/docs/2.2.1.GA/userguide/html/ExceptionHandling.html#builtinException>`_
+that need to be handled and reformatted before sending back a response. The ErrorCode on the
+Failure exception will be used as the status of the response as long as it is a valid code; if the
+code is invalid, this will be treated as an Internal Server Error (500). The response message will
+be derived from the resulting status code.
+
+ThrowableExceptionMapper is the exception mapper that will behave as a catch-all for all exceptions
+that go unhandled. An unhandled exception could match any of the following scenarios:
+
+- The exception was a checked exception that was not handled by try/catch blocks.
+- The exception was a checked exception that was missing an exception mapper or the exception mapper was implemented incorrectly.
+- The exception was an unchecked exception.
+
+Any exception thrown in a resource matching one of the scenarios mentioned above will be treated as
+an Internal Server Error (500).
+
+Keep in mind that an unhandled exception will be caught by the exception mapper that is most
+specific to the exception. If there is not an exception mapper for the specific exception, the next
+closest exception mapper in the exception's hierachy will handle it. For example, if a
+`NotFoundException <http://docs.oracle.com/javaee/7/api/javax/ws/rs/NotFoundException.html>`_
+(which extends from WebApplicationException) is thrown, the WebApplicationExceptionMapper will
+handle it, rather than the ThrowableExceptionMapper.
+
+Implementation
+______________
+
+Each of the exception mappers mentioned above are readily available for consumption. They
+are automatically added to a project that have installed either the BeadledomModule
+(WebApplicationExceptionMapper and ThrowableExceptionMapper only) or the ResteasyModule
+(WebApplicationExceptionMapper, FailureExceptionMapper, and ThrowableExceptionMapper).
+
+NOTE: If your service is able to throw custom exceptions, it is suggested that a custom
+ExceptionMapper is added to your project as well. This will allow you to customize the response that
```

Comment:
Can we make `ExceptionMapper` link here https://docs.jboss.org/resteasy/docs/3.1.4.Final/userguide/html/ExceptionHandling.html#ExceptionMappers ?

Reply:
Should we mention to stay consistent with the rest of the error handling `JsonError` should be used for the entity?

Reply:
https://github.com/cerner/beadledom/pull/45/commits/2530c00610b01c4648943d907f7d06846a37a0a4#diff-2275a5c50f335dc3eeff64d39ae09f45R117